module-2/backend/week_one.md

```diff
@@ -4,31 +4,90 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+GET - Gets what you are asking it to<br>
+POST - Creates a new thing of what you asked it to<br>
+DELETE - Deletes what you ask it to<br>
+PATCH - Replaces part of what you are asking it to<br>
+PUT - Replaces ALL of what you are asking it to<br>
+
 2. What is Sinatra?
+A framework for creating web applications
+
 4. What is MVC?
+Model, Views, Controller
+A structure for how you view the pages, how the pages talk to eachother and how the database is manipulated.
+
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+So we always have a basic framework to work off of every time we start.
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+Instance variables
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+  @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+
+```ruby
+  get '/horses' do
+  :locals {:mr.ed => mr.ed}
+  end
+  ```
+ 
 9. What's the purpose of ERB?
+So you can render ruby in HTML
+
 10. Why do I need a development AND test database?
+Development you can change as you go. Test would be like a fixture that you can test for specific things.
+
 11. What's responsive design?
+When you write your code so that people can manipulate the size of your page and it can keep all the content separate.
+
 12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
+Create, Read, Update, Delete
+It's basically how every website that you can actually do anything on works.
+
+13. What does HTTP stand for?
+Hypertext Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<% %> - will not render on the page
+<%= %> - renders on the page
+
 15. What's an ORM?
+Object Relational Mapper
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+ActiveRecord
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
+READ - GET - view all<br>
+READ - GET - view one<br>
+CREATE - GET - form for new thing<br>
+CREATE - POST - creating the new thing<br>
+UPDATE - GET - form for editing the thing<br>
+UPDATE - PUT/PATCH - updating the thing<br>
+DELETE - DESTROY - deleting the thing<br>
+
+
+18. What's a migration?
+Where your table is created and attributes are assigned.
+
 19. When you create a migration, does it automatically modify your database?
+No. You create the migration. Then inside you put your validations and relationships then you use rake db:migrate to update your database schema.

```

Comment:
Validations/relationships will go in your model. The column/table names go into your migration.